pkg/cloudprovider/providers/azure/azure_storageaccount.go

```diff
@@ -75,3 +79,52 @@ func (az *Cloud) getStorageAccesskey(account string) (string, error) {
 	}
 	return "", fmt.Errorf("no valid keys")
 }
+
+// SearchStorageAccount search storage account, create one storage account(with genAccountNamePrefix) if not found, return accountName, accountKey
+func (az *Cloud) SearchStorageAccount(accountName, accountType, location, genAccountNamePrefix string) (string, string, error) {
```

Comment:
This func is to be used in package only, consider making it lower case

Also 'search' looks like a read only operation, may be use 'ensure' better, like other places

Reply:
fixed, I am also considering a good name, ensureStorageAccount is better